src/_includes/components/navigation.liquid

```diff
@@ -46,11 +46,7 @@
 
   navToggler.addEventListener('click', () => {
     // Toggle the 'hidden' class on the navigation links container
-    navigationContainer.classList.toggle('invisible');
+    navigationContainer.classList.toggle('hidden');
```

Comment:
Yeah, I was unsure where to place the script with Eleventy. Do you think this would be the best approach?
https://www.11ty.dev/docs/data-js/

Reply:
I don't think so.  That looks like it's for using JS to load data.  I think you want to use JS templates: https://www.11ty.dev/docs/assets/

Click on "Customize for: JavaScript".

Reply:
Ah okay, So I moved the navigation script to an external `JS` file which is located in the `scripts` folder under the filename `navigation.js` It is being called in the `navigation.liquid` partial in the form of `<script src="/navigation.js"></script> at the end of the file. Is this a better practice? or should it be called in the `base.liquid` template?
`

Reply:
I think you could do it this way, but if you want to have modular JS, you'll probably want to let eleventy bundle it.  You would create a `js-bundle.njk` file and in your `script` tag you'd set the `src="bundle.js"`.  Then in your `js-bundle.njk` file, include the navigation.js  Right now your JS is simple, but if later you needed to use NPM modules or create your own modules, you'll want to bundle it with eleventy, might as well create the bundle now.

Reply:
Where would the `js-bundle.njk` file live? In the root?

Reply:
Hm, I currently have a `bundle.js` file in the root with the javascript for the navigation and its being passed through in the `eleventy.js` config file

```
module.exports = function(eleventyConfig) {
	eleventyConfig.addPassthroughCopy("bundle.js");
};
```
and it's being called as such

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Digital Services Coalition</title>
    <link
      rel="stylesheet"
      type="text/css"
      href="{{ '/styles/main.css' | url }}" />
    <link
      rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/material-design-icons/3.0.1/iconfont/material-icons.min.css"
      integrity="sha256-x8PYmLKD83R9T/sYmJn1j3is/chhJdySyhet/JuHnfY="
      crossorigin="anonymous" />
    <link
      rel="icon"
      href="/favicon/favicon.ico"
      type="image/x-icon">
  </head>
  <body>
    {% include "components/navigation.liquid" %}
    <main>
      {{ content }}
    </main>
    <div class="flex flex-col">
      {% include "components/navigation.liquid" %}
      <a
        href="https://github.com/DSCoalition/digitalservicescoalition.org"
        target="_blank"
        class="text-end mr-5 mb-5 desktop:mr-20 desktop:mb-10">
        <p class="text-grayLight text-xs">Our code is open source.</p>
      </a>
    </div>
    <script src="/bundle.js"></script>
    <script src="{{ '/assets/main.js' | url }}"></script>
  </body>
</html>
```

Reply:
I believe so.  I've never actually used 11ty.  I'm just reading the documentation and it doesn't specify a path.  I would say try adding it at the root and run `npm build` to see if it generates the bundle.js file.

Reply:
So without the `js-bundle.njk` file it does create a `bundle.js ` file in the output folder in the root

Reply:
So, if you want to just use the passthrough method.  You wouldn't create a `js-bundle.njk` file.  That file is just for the template method.

Reply:
Ah okay, so it should be fine as is at the moment without the `js-bundle.njk` and just using the passthrough method.

Reply:
Yeah, that should be fine for now.  It will basically just copy the files to the build output folder. It looks like it will also let you include node modules if you need them as well.  So, it should be enough for now.

Reply:
Awesome, thanks, Tri! That was super helpful!